src/hooks/useFileSystemWatcher.tsx

```diff
@@ -11,7 +11,7 @@ type Path = string
 // watcher.addListener(() => { ... }).
 
 export const useFileSystemWatcher = (
-  callback: (path: Path) => void,
+  callback: (path: Path) => any | Promise<any>,
```

Comment:
I don't think there's any need for this change. It disables the type checker for no reason.
```suggestion
  callback: (path: Path) => void,
```

Reply:
There is, go and try it with void :)

Reply:
Admittedly it was pointless of me to do any | Promise<any>. I should have just been any...

Reply:
I think it's silly to assume `void` is correct on behalf of the user. We don't care what the callback returns. Hm. I wonder if maybe <T>(path) => T would have worked

Reply:
Ok after like another 5 seconds I'm telling myself there is no way to get the return value anyway. I should have just done void | Promise<void> :facepalm:

Reply:
As long as there's no `any`, I'm happy 🙂

Reply:
Actually it doesn't work with `void | Promise<void>` either. The linter is dumb.

![out](https://github.com/user-attachments/assets/8a1fe195-bf94-40df-bf09-16fb4d3286b3)

Reply:
#4098